packages/@sanity/core/src/util/checkRequiredDependencies.ts

```diff
@@ -0,0 +1,238 @@
+    )
+  ) {
+    output.warn(oneline`
+      Declared version of styled-components (${declaredStyledComponentsVersion})
+      is not compatible with the version required by @sanity/base (${wantedStyledComponentsVersionRange}).
+      This might cause problems!
+    `)
+  }
+
+  // Ensure the studio has _installed_ a version of `styled-components`
+  if (!installedStyledComponentsVersion) {
+    throw new Error(oneline`
+      Declared dependency \`styled-components\` is not installed - run
+      \`npm install\` or \`yarn\` to install it before re-running this command.
+    `)
+  }
+
+  // The studio should have an _installed_ version of `styled-components`, and it should
+  // be semver compatible with the version specified in `@sanity/base` peer dependencies.
+  if (!semver.satisfies(installedStyledComponentsVersion, wantedStyledComponentsVersionRange)) {
+    output.warn(oneline`
+      Installed version of styled-components (${installedStyledComponentsVersion})
+      is not compatible with the version required by @sanity/base (${wantedStyledComponentsVersionRange}).
+      This might cause problems!
+    `)
+  }
+}
+
+/**
+ * Read the declared peer dependencies of the installed `@sanity/base` module
+ *
+ * @param studioPath Path to the studio, in order to resolve `@sanity/base`
+ * @returns Object of peer dependencies (name => version), if any
+ */
+async function readBasePeerDependencies(
+  studioPath: string
+): Promise<Record<string, string | undefined>> {
+  const manifestPath = resolveFrom.silent(studioPath, path.join('@sanity/base', 'package.json'))
+
+  // If we can't resolve the manifest path, that means `@sanity/base` is not installed
+  if (!manifestPath) {
+    throw new Error(
+      'Failed to resolve @sanity/base/package.json - install dependencies with `npm install` or `yarn`.'
+    )
+  }
+
+  const {peerDependencies} = await readPackageManifest(manifestPath)
+  return peerDependencies
+}
+
+/**
+ * Reads the version number of the _installed_ module, or returns `null` if not found
+ *
+ * @param studioPath Path of the studio
+ * @param moduleName Name of module to get installed version for
+ * @returns Version number, of null
+ */
+async function readModuleVersion(studioPath: string, moduleName: string): Promise<string | null> {
+  const manifestPath = resolveFrom.silent(studioPath, path.join(moduleName, 'package.json'))
+  return manifestPath ? (await readPackageManifest(manifestPath)).version : null
+}
+
+/**
+ * Read the `package.json` file at the given path and return an object that guarantees
+ * the presence of name, version, dependencies, dev dependencies and peer dependencies
+ *
+ * @param packageJsonPath Path to package.json to read
+ * @returns Reduced package.json with guarantees for name, version and dependency fields
+ */
+async function readPackageManifest(packageJsonPath: string): Promise<PackageManifest> {
+  let manifest: unknown
+  try {
+    manifest = await readJSON(packageJsonPath)
```

Comment:
i wonder if `require(packageJsonPath)` would be better. might be more compatible with node e.g. if someone had registered a require hook in there or something. in theory resolveFrom should catch that though